core/src/main/java/com/amplifyframework/devmenu/DevMenuLogsFragment.java

```diff
@@ -39,10 +46,23 @@ public DevMenuLogsFragment() { }
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        View logsView = inflater.inflate(R.layout.dev_menu_fragment_logs, container, false);
+        logsView = inflater.inflate(R.layout.dev_menu_fragment_logs, container, false);
         // Display the logs (if any).
-        TextView logsText = logsView.findViewById(R.id.logs_text);
-        logsText.setText(DeveloperMenu.singletonInstance(getContext()).getLogs());
+        developerMenu = DeveloperMenu.singletonInstance(getContext());
+        logsText = logsView.findViewById(R.id.logs_text);
+        logsText.setText(developerMenu.getLogs());
+        // Search the logs when the search button is pressed.
+        logsView.findViewById(R.id.search_logs_button).setOnClickListener(view -> searchLogs());
```

Comment:
Instead of having an explicit button, could you just display results in real time, updating them each time a new character is entered in the search bar?    If you use a SearchView, you could implement this with the onQueryTextChanged() method in [OnQueryTextListener](https://developer.android.com/reference/android/widget/SearchView.OnQueryTextListener).